packages/legacy/core/App/services/keychain.ts

```
@@ -142,22 +143,30 @@ export const loadWalletKey = async (title?: string, description?: string): Promi
 export const loadWalletSecret = async (
   title?: string,
   description?: string
-): Promise<{ secret: WalletSecret | undefined; err: string }> => {
+): Promise<WalletSecret | undefined> => {
   let salt: WalletSalt | undefined
   let key: WalletKey | undefined
-  let err = ''
+  let secret: WalletSecret | undefined
   try {
     salt = await loadWalletSalt()
     key = await loadWalletKey(title, description)
   } catch (e: any) {
-    err = e?.message ?? e
+    // should this throw instead?
+    DeviceEventEmitter.emit(EventTypes.BIOMETRY_ERROR, e?.message ?? e)
```

The original code captured this error and added it to the return. It emitted an error but continued on with an undefined wallet secret, I'm wondering if this should throw an error instead emitting an error

---

That event will trigger a global error catcher but if the wallet can't be unlocked the entire app is pretty useless. My though is no point in carrying on with an invalid secret.